### ISSUE TITLE
RStudio IDE could preview equations introduced by a \begin{} block not just $$

### ISSUE THREAD

**Issue by howthebodyworks** [rstudio/rstudio#5264]
[As mentioned in the community forums](https://community.rstudio.com/t/how-to-get-inline-preview-of-numbered-math-equations-using-begin-equation/37263), There iis a pain point with using Rstudio to construct articles and blog post with RMarkdown and bookdown.

To wit, RStudio IDE supports inline/tooltip [mathjax equation previews of equations using $ or $$ syntax](https://community.rstudio.com/t/what-is-rstudios-roadmap-for-latex-enhancements/846/2) for RMarkdown documents, e.g. the following markup will be rendered as an inline preview. 

```markdown
$$
a^2+b^2=c^2
$$
```

FWIW I think this is the best way of doing this and this feature is genius :1st_place_medal: )

However, [if I want numbered labeled equations in bookdown](https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html#equations) I need to use LaTeX equation syntax.

```markdown
\begin{equation}
a^2+b^2=c^2 \label{eq:pythagoras}
\end{equation}
```

(Typically actually `\begin{align}` for me)
These are not, AFAICT, picked up by the parser and previewed in RStudio; you have to render an HTML or PDF output to see them. This is vexing because it is precisely these equations which are likely to be complex enough to need preview.

Ideally, previews of such equations should also work [inside a theorem](https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html#theorems), although I could survive without that.

`````markdown
```{theorem, pyth, name="Pythagorean theorem"}
For a right triangle, if $c$ denotes the length of the hypotenuse
and $a$ and $b$ denote the lengths of the other two sides, we have

\begin{equation}a^2 + b^2 = c^2\end{equation}
```
`````

**Comment by howthebodyworks** [comment]
I presume this is the right place to open this request, although clearly it relates also to @yihui et al's  [bookdown](https://github.com/rstudio/bookdown), which already has [at least one other request](https://github.com/rstudio/bookdown/issues/689) for (AFAICT) Rstudio IDE preview improvements. :wink:

**Comment by ronblum** [comment]
@howthebodyworks If @kevinushey says this is the place to put it, then it's the right place :)  Thank you for bring this over from the Community forum and for providing this great detail and reference links. We'll review the request in our ongoing development of RStudio.

**Comment by danmackinlay** [comment]
I'm not sure if this is the right thread to relay this, but note that previews of equations inside theorems also seems broken, in that there is no equation preview even of the `$a$` inline equations in the theorem environments in RStudio – _no_ part of the following theorem is previewed in the editor.

````markdown
```{theorem, pyth, name="Pythagorean theorem"}
For a right triangle, if $c$ denotes the length of the hypotenuse
and $a$ and $b$ denote the lengths of the other two sides, we have

\begin{equation}a^2 + b^2 = c^2\end{equation}
```
````

**Comment by yihui** [comment]
@danmackinlay That is expected, because the math expression is essentially inside a code block, although it is a special code block in **bookdown**. In the future, I'll provide an alternative syntax to write theorems in **bookdown**, which will solve your problem of previewing math expressions in `$ $`, but the original problem (i.e. previewing equations in the `equation` environment) won't be solved before the proper IDE support comes.

**Comment by howthebodyworks** [comment]
This issue is also notable for `knitr` documents in LaTeX, or, I suppose LaTeX documents generally - it would be amazing if the RStudio parser could preview general equations in LaTeX document rather than having to knit the whole PDF. I am aware than this could get crazy, as people creating LaTeX documents often have very serious macros and exotic LaTeX environment etc which are not the main focus RStudio IDE. Nonetheless for the kind of documents I write (IEEE stuff where we are not permitted to get too fancy) it would be very useful and I imagine the parsing challenge is similar to the RMarkdown case. (Although I have not in fact looked at the RStudio code :wink: )

**Comment by yihui** [comment]
FYI, the alternative syntax I mentioned above has been provided in **bookdown**: https://blog.rstudio.com/2020/12/21/rmd-news/#a-new-way-to-use-theorem-and-proof-environments

**Comment by InazeaAnazasi** [comment]
@yihui is there a similar method to preview formulas inside environments like `align`? My workaround is like this:

$$\begin{align*}
x&=4\\
y&=5
\end{align*}$$

 which allows for previews, but (obviously) runs into troubles when knitting to PDF.

**Comment by yihui** [comment]
@InazeaAnazasi No. You are requesting the same feature as the OP, which has not been implemented. I was only replying to @howthebodyworks and @danmackinlay above regarding a side issue in **bookdown**.

**Comment by EvgeniOvcharov** [comment]
I have recently started using rmarkdown and bookdown to write lecture notes for a statistical course. This requires using more advanced LaTex functionality such as numbered equations, theorem environments, latex shortcuts, installing additional packages, etc. There are many issues with these features. I would also like to be able to preview numbered equations. Hope that Rstudio keeps improving its functionality.

**Comment by ecceeecs** [comment]
New markdown user, running into the same issue as well. A pain point indeed. Does anyone involved in this discussion have a workaround/compromise they'd like to share?
